server/blob/blob-objectstorage/src/main/java/org/apache/james/blob/objectstorage/AESPayloadCodec.java

```diff
@@ -71,7 +83,9 @@ public Payload write(InputStream is) {
     @Override
     public InputStream read(Payload payload) throws IOException {
         try {
-            return streamingAead.newDecryptingStream(payload.openStream(), PBKDF2StreamingAeadFactory.EMPTY_ASSOCIATED_DATA);
+            byte[] ciphertext = IOUtils.toByteArray(payload.getPayload().openStream());
+            byte[] decrypt = aead.decrypt(ciphertext, EMPTY_ASSOCIATED_DATA);
```

Comment:
we should at least use a `FileBackedOutputStream` to avoid killing the server with allocation (waiting for streaming implementation)